extensions/amp-animation/0.1/amp-animation.js

```diff
@@ -42,17 +60,125 @@ export class AmpAnimation extends AMP.BaseElement {
     user().assert(isExperimentOn(this.win, TAG),
         `Experiment "${TAG}" is disabled.`);
 
+    // Trigger.
+    const trigger = this.element.getAttribute('trigger');
+    if (trigger) {
+      this.triggerOnVisibility_ = user().assert(
+          trigger == 'visibility',
```

Comment:
As discussed offline, we will extend the visibility to the direct parent, but for now we will keep it defined as `:root` == embed visibility. As such `amp-animation` is restricted to only be placed directly inside `<body>` element.